data/fwupdagent.service

```diff
@@ -0,0 +1,14 @@
+[Unit]
+Description=Firmware update daemon agent
+Documentation=https://fwupd.org/
+Requires=fwupd.service
+After=fwupd.service
+
+[Service]
+Type=oneshot
+User=nobody
+ExecStart=@libexecdir@/fwupd/fwupdagent
+PrivateTmp=yes
+ProtectHome=yes
+ProtectSystem=full
+SystemCallFilter=~@mount
```

Comment:
Thoughts on adding these too?
`RestrictRealtime`
`MemoryDenyWriteExecute`
`ProtectKernelModules`
`ProtectControlGroups`

If they're added probably need to do them dynamically like is done for `fwupd.service.in` too though to let fwupd keep working with older systemd for RHEL7/CentOS 7.

Reply:
All good ideas, I just copied the logic we use for `fwupd.service` minus the `ReadWritePaths`. Thanks.